support-frontend/assets/components/subscriptionCheckouts/summary.tsx

```diff
@@ -1,41 +1,113 @@
-import { Component } from 'react';
-import type { $Call } from 'utility-types';
-import type { ProductPrice } from 'helpers/productPrice/productPrices';
+import { useState } from 'react';
 import 'helpers/productPrice/productPrices';
-import type { GridImg } from 'components/gridImage/gridImage';
 import { PriceLabel } from 'components/priceLabel/priceLabel';
 import type { BillingPeriod } from 'helpers/productPrice/billingPeriods';
+import type { ProductPrice } from 'helpers/productPrice/productPrices';
 import 'components/gridImage/gridImage';
 import type { Promotion } from 'helpers/productPrice/promotions';
 import { getAppliedPromo, hasDiscount } from 'helpers/productPrice/promotions';
 import { GuardianWeekly } from 'helpers/productPrice/subscriptions';
 import type { SubscriptionProduct } from 'helpers/productPrice/subscriptions';
 import SvgDropdownArrowUp from './dropDownArrowUp.svg';
-import styles from './summary.module.scss';
+import moduleStyles from './summary.module.scss';
 
-type GridImageType = typeof import('components/gridImage/gridImage').default;
-// Types
-export type DataListItem = {
+// Type declarations
+type DataListItem = {
 	title: string;
 	value: string;
 };
-type PropTypes = {
+
+type SummaryPropTypes = {
 	billingPeriod: BillingPeriod;
 	changeSubscription?: string | null;
 	dataList: DataListItem[];
-	description: string | null | undefined;
-	image: $Call<GridImageType, GridImg>;
+	description?: string | null;
+	image: JSX.Element | null;
 	productPrice: ProductPrice;
 	title: string;
-	// eslint-disable-next-line react/no-unused-prop-types
 	product: SubscriptionProduct;
 	orderIsAGift?: boolean;
 };
-type StateTypes = {
+
+type TabletAndDesktopPropTypes = {
+	billingPeriod: BillingPeriod;
+	changeSubscription?: string | null;
+	dataList: DataListItem[];
+	description?: string | null;
+	image: JSX.Element | null;
+	productPrice: ProductPrice;
+	title: string;
+	product: SubscriptionProduct;
+	orderIsAGift?: boolean;
+};
+
+type MobilePropTypes = {
+	billingPeriod: BillingPeriod;
+	changeSubscription?: string | null;
+	description: string | null | undefined;
+	productPrice: ProductPrice;
+	title: string;
+	showDropDown: boolean;
+	onClick: () => void;
+	deliveryMethod?: string;
+	paper: boolean;
+};
+
+type ShowDropDownPropTypes = {
+	billingPeriod: BillingPeriod;
+	changeSubscription?: string | null;
+	description?: string | null;
+	productPrice: ProductPrice;
+	title: string;
+	onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+	deliveryMethod?: string;
 	showDropDown: boolean;
 };
 
-// Helpers
+type HideDropDownPropTypes = {
+	billingPeriod: BillingPeriod;
+	productPrice: ProductPrice;
+	title: string;
+	onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
+	paper: boolean;
+	showDropDown: boolean;
+};
+
+// Variable declarations
+const styles = moduleStyles as {
```

Comment:
I followed this pattern of using a Type assertion for the styles in a `.scss` module as I've seen it elsewhere, it's not ideal as it it makes this object the source of truth for what styles are available, rather than the `.scss` module. Ideally we'd convert all these over to emotion `css` variables.

Reply:
Yeah i'm inclined to say we leave it for now and wait to migrate to emotion, this styles object does at least provide a quick overview of all the styles actually used in the module for whoever picks up the migration.

Reply:
Yeah makes sense 👍